### DISCUSSION TITLE
What about letting the VM decide whether to JIT or not?

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#403] [General]
At least, this is how they are doing it on the JVM and .NET runtime.

The original idea is here: https://github.com/objeck/objeck-lang/issues/387#issuecomment-1817046516

You are placing too much burden on the developers, as they have to decide what to be JIT-ed. With my very limited knowledge, I still know it must take serious profiling to know what the bottleneck is. Do you offer such a profiling tool? Even if you do, the average Joe like me will not be able to take advantage of it.

**Comment by objeck** [comment]
Ok, however, I enjoy working more on the VM vs. the frontend language. Maybe, this is the lack of features on the front end. Maybe, Objeck just becomes an OK VM and we drop the language frontend?
